Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -247,10 +249,26 @@ extension ActivityStreamPanel {
         }
     }
 
+    /*
+     sites, defaultsites
+     take sites and map replacing any that match defaultsites
+     filter the defaultsites to remove any dupes. 
+     append the defaultsites to the sites
+ */
+
     private func reloadTopSites() {
         invalidateTopSites().uponQueue(dispatch_get_main_queue()) { result in
-            let sites = result.successValue ?? []
-            self.topSites = sites
+            // Merge default topsites with a user's topsites. Replace any topsites that are also in the default topsites
+            let defaultSites = self.defaultTopSites()
+            let sites = (result.successValue ?? []).map { site in
+                return defaultSites.find { $0.tileURL.extractDomainName() == site.tileURL.extractDomainName() } ?? site
```

Comment:
Unfortunately the `extractDomainName` can be _really_ slow and calling it every map loop and every find lookup can be costly. Might make sense to calculate the domain names and use that instead.